File: testdata/jennies/builders/factories/PHPBuilder/src/Promql/FuncCallExprBuilder.php
<?php

namespace Grafana\Foundation\Promql;

/**
 * @implements \Grafana\Foundation\Cog\Builder<\Grafana\Foundation\Promql\FuncCallExpr>
 */
class FuncCallExprBuilder implements \Grafana\Foundation\Cog\Builder
{
    protected \Grafana\Foundation\Promql\FuncCallExpr $internal;

    public function __construct()
    {
    	$this->internal = new \Grafana\Foundation\Promql\FuncCallExpr();
    $this->internal->type = "funcCallExpr";
    }

    /**
     * Builds the object.
     * @return \Grafana\Foundation\Promql\FuncCallExpr
     */
    public function build()
    {
        return $this->internal;
    }

    public function function(string $function): static
    {
        if (!(strlen($function) >= 1)) {
            throw new \ValueError('strlen($function) must be >= 1');
        }
        $this->internal->function = $function;
    
        return $this;
    }

    /**
     * @param array<\Grafana\Foundation\Cog\Builder<\Grafana\Foundation\Promql\Expr>> $args
     */
    public function args(array $args): static
    {
            $argsResources = [];
            foreach ($args as $r1) {
                    $argsResources[] = $r1->build();
            }
        $this->internal->args = $argsResources;
    
        return $this;
    }

    /**
     * Modified by veneer 'Duplicate[args]'
     * Modified by veneer 'ArrayToAppend'
     * @param \Grafana\Foundation\Cog\Builder<\Grafana\Foundation\Promql\Expr> $arg
     */
    public function arg(\Grafana\Foundation\Cog\Builder $arg): static
    {
        $argResource = $arg->build();
        $this->internal->args[] = $argResource;
    
        return $this;
    }

}

File: testdata/jennies/builders/factories/CSharpBuilder/src/Grafana/Foundation/Promql/FuncCallExprBuilder.cs
namespace Grafana.Foundation.Promql;


public class FuncCallExprBuilder : Cog.IBuilder<FuncCallExpr>
{
    protected readonly FuncCallExpr @internal;

    public FuncCallExprBuilder()
    {
        this.@internal = new FuncCallExpr();
        this.@internal.Type = "funcCallExpr";
    }

    public FuncCallExprBuilder Function(string function)
    {
        if (!(function.Length >= 1))
        {
            throw new System.ArgumentException("function.Length must be >= 1");
        }
        this.@internal.Function = function;
        return this;
    }

    public FuncCallExprBuilder Args(List<Cog.IBuilder<Expr>> args)
    {
        List<Expr> argsResources = new List<Expr>();
        foreach (Cog.IBuilder<Expr> r1 in args)
        {
                Expr argsDepth1 = r1.Build();
                argsResources.Add(argsDepth1);
        }
        this.@internal.Args = argsResources;
        return this;
    }

    public FuncCallExprBuilder Arg(Cog.IBuilder<Expr> arg)
    {
        if (this.@internal.Args == null)
        {
            this.@internal.Args = new List<Expr>();
        }
        Expr argResource = arg.Build();
        this.@internal.Args.Add(argResource);
        return this;
    }

    public FuncCallExpr Build()
    {
        return this.@internal;
    }
}
